File: ios-catalogue/App/Shell/DetailPane.swift
import SwiftUI
import CoreUI
import Components

struct DetailPane: View {
    let entry: CatalogueEntry?

    @Environment(\.colorScheme) private var scheme

    var body: some View {
        ZStack {
            PrismaSemanticColors.surfaceBase.themed(scheme).ignoresSafeArea()
            if let entry {
                entryContent(entry)
            } else {
                emptyContent
            }
        }
    }

    private var emptyContent: some View {
        let total = CatalogueRegistry.entries.count
        let implemented = 7   // 6 foundations + Button — keep in sync with implementation phases

        return VStack(spacing: PrismaSpacing.sp5) {
            Spacer()
            // Decorative accent ring.
            Circle()
                .fill(PrismaSemanticColors.accentSubtle.themed(scheme))
                .frame(width: 72, height: 72)

            Text("Prisma")
                .font(PrismaTypography.displaySm.font)
                .foregroundStyle(PrismaSemanticColors.textPrimary.themed(scheme))

            Text("Pick a component from the sidebar to see its showcase.")
                .font(PrismaTypography.bodyLg.font)
                .foregroundStyle(PrismaSemanticColors.textSecondary.themed(scheme))
                .multilineTextAlignment(.center)

            HStack(spacing: PrismaSpacing.sp7) {
                stat(label: "ENTRIES", value: "\(total)")
                stat(label: "IMPLEMENTED", value: "\(implemented)")
                stat(label: "PHASE", value: "1")
            }
            .padding(.top, PrismaSpacing.sp4)
            Spacer()
        }
        .padding(PrismaSpacing.sp7)
        .frame(maxWidth: .infinity, maxHeight: .infinity)
    }

    @ViewBuilder
    private func stat(label: String, value: String) -> some View {
        VStack(spacing: PrismaSpacing.sp1) {
            Text(value)
                .font(.system(size: 26, weight: .semibold, design: .monospaced))
                .foregroundStyle(PrismaSemanticColors.accentDefault.themed(scheme))
            Text(label)
                .font(.system(size: 11, weight: .semibold))
                .foregroundStyle(PrismaSemanticColors.textTertiary.themed(scheme))
        }
    }

    private func entryContent(_ entry: CatalogueEntry) -> some View {
        // The SDUI demo is meant to look like a real product screen — no
        // breadcrumb, no entry-title chrome, no catalogue horizontal padding.
        // The JSON document owns its own padding via the root column's
        // `padding` property, so we render edge-to-edge and let the JSON
        // dictate inset.
        if entry.key == "sdui.live" {
            return AnyView(
                ScrollView {
                    showcase(for: entry)
                        .frame(maxWidth: .infinity, alignment: .leading)
                }
            )
        }
        return AnyView(ScrollView {
            VStack(alignment: .leading, spacing: PrismaSpacing.sp4) {
                // Section breadcrumb pill.
                Text(entry.section.rawValue)
                    .font(.system(size: 11, weight: .semibold))
                    .foregroundStyle(PrismaSemanticColors.textSecondary.themed(scheme))
                    .padding(.horizontal, PrismaSpacing.sp3)
                    .padding(.vertical, PrismaSpacing.sp1)
                    .background(PrismaSemanticColors.surfaceSunken.themed(scheme))
                    .clipShape(Capsule())

                Text(entry.title)
                    .font(PrismaTypography.displaySm.font)
                    .foregroundStyle(PrismaSemanticColors.textPrimary.themed(scheme))

                if !hasShowcase(entry) {
                    Text("Spec ready · implementation pending in upcoming phase.")
                        .font(PrismaTypography.bodyMd.font)
                        .foregroundStyle(PrismaSemanticColors.textTertiary.themed(scheme))
                }

                Rectangle()
                    .fill(PrismaSemanticColors.borderSubtle.themed(scheme))
                    .frame(height: 1)
                    .padding(.vertical, PrismaSpacing.sp1)

                if hasShowcase(entry) {
                    showcase(for: entry)
                        .padding(.top, PrismaSpacing.sp4)
                } else {
                    placeholderState(entry: entry)
                }

                Spacer(minLength: PrismaSpacing.sp7)
            }
            .padding(.horizontal, PrismaSpacing.sp8)
            .padding(.vertical, PrismaSpacing.sp7)
            .frame(maxWidth: .infinity, alignment: .leading)
        })
    }

    private func hasShowcase(_ entry: CatalogueEntry) -> Bool {
        // Every entry in the registry has a real showcase now.
        return true
    }

    @ViewBuilder
    private func placeholderState(entry: CatalogueEntry) -> some View {
        VStack(alignment: .leading, spacing: PrismaSpacing.sp3) {
            Text("Implementation queued")
                .font(PrismaTypography.titleMd.font)
                .foregroundStyle(PrismaSemanticColors.textPrimary.themed(scheme))

            Text("The spec for \(entry.title.lowercased()) is complete. Component code lands in the upcoming phase per docs/TODO.md.")
                .font(PrismaTypography.bodyMd.font)
                .foregroundStyle(PrismaSemanticColors.textSecondary.themed(scheme))

            if !entry.tags.isEmpty {
                HStack(spacing: PrismaSpacing.sp2) {
                    ForEach(Array(entry.tags.prefix(4)), id: \.self) { tag in
                        Text(tag)
                            .font(.system(size: 11, weight: .semibold))
                            .foregroundStyle(PrismaSemanticColors.textSecondary.themed(scheme))
                            .padding(.horizontal, PrismaSpacing.sp3)
                            .padding(.vertical, PrismaSpacing.sp1)
                            .background(PrismaSemanticColors.surfaceRaised.themed(scheme))
                            .clipShape(Capsule())
                    }
                }
            }
        }
        .padding(PrismaSpacing.sp7)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(PrismaSemanticColors.surfaceSunken.themed(scheme))
        .clipShape(RoundedRectangle(cornerRadius: PrismaRadius.lg))
    }

    @ViewBuilder
    private func showcase(for entry: CatalogueEntry) -> some View {
        switch entry.key {
        // Foundations
        case "foundation.typography": TypographyShowcase()
        case "foundation.colors":     ColorShowcase()
        case "foundation.spacing":    SpacingShowcase()
        case "foundation.radius":     RadiusShowcase()
        case "foundation.elevation":  ElevationShowcase()
        case "foundation.motion":     MotionShowcase()
        case "foundation.icons":      IconsShowcase()
        // Inputs
        case "input.button":            ButtonShowcase()
        case "input.textfield":         TextFieldShowcase()
        case "input.checkbox":          CheckboxShowcase()
        case "input.radio":             RadioShowcase()
        case "input.switch":            SwitchShowcase()
        case "input.slider":            SliderShowcase()
        case "input.segmentedControl":  SegmentedControlShowcase()
        case "input.searchBar":         SearchBarShowcase()
        case "input.autocomplete":      AutocompleteShowcase()
        case "input.stepper":           StepperShowcase()
        case "input.tagInput":          TagInputShowcase()
        case "input.datePicker":        DatePickerShowcase()
        case "input.timePicker":        TimePickerShowcase()
        case "input.colorPicker":       ColorPickerShowcase()
        // Feedback
        case "feedback.toast":       ToastShowcase()
        case "feedback.banner":      BannerShowcase()
        case "feedback.modal":       ModalShowcase()
        case "feedback.bottomSheet": BottomSheetShowcase()
        case "feedback.popover":     PopoverShowcase()
        case "feedback.tooltip":     TooltipShowcase()
        case "feedback.loading":     LoadingShowcase()
        case "feedback.skeleton":    SkeletonShowcase()
        case "feedback.badge":       BadgeShowcase()
        case "feedback.emptyState":  EmptyStateShowcase()
        case "feedback.drawer":      DrawerShowcase()
        // Navigation
        case "navigation.tabs":            TabsShowcase()
        case "navigation.chip":            ChipShowcase()
        case "navigation.commandPalette":  CommandPaletteShowcase()
        case "navigation.pagination":      PaginationShowcase()
        case "navigation.breadcrumb":      BreadcrumbShowcase()
        case "navigation.wizard":          WizardShowcase()
        // Data display
        case "data.card":        CardShowcase()
        case "data.listItem":    ListItemShowcase()
        case "data.avatar":      AvatarShowcase()
        case "data.avatarGroup": AvatarGroupShowcase()
        case "data.divider":     DividerShowcase()
        // Server-driven
        case "sdui.live":        SDUIScreen()
        default:
            Text("Phase 0 placeholder — implementation lands per-phase per docs/TODO.md.")
                .font(PrismaTypography.bodyMd.font)
                .foregroundStyle(PrismaSemanticColors.textSecondary.themed(scheme))
        }
    }
}

File: android-catalogue/catalogue/src/main/java/xyz/ksharma/prisma/catalogue/shell/DetailPane.kt
package xyz.ksharma.prisma.catalogue.shell

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.unit.dp
import xyz.ksharma.prisma.catalogue.registry.CatalogueEntry
import xyz.ksharma.prisma.catalogue.registry.CatalogueRegistry
import xyz.ksharma.prisma.coreui.themed
import xyz.ksharma.prisma.tokens.PrismaRadius
import xyz.ksharma.prisma.tokens.PrismaSemanticColors
import xyz.ksharma.prisma.tokens.PrismaSpacing
import xyz.ksharma.prisma.tokens.PrismaTypography

@Composable
public fun DetailPane(
    entry: CatalogueEntry?,
    scrollOffsets: androidx.compose.runtime.snapshots.SnapshotStateMap<String, Int>,
    modifier: Modifier = Modifier,
) {
    Box(
        modifier = modifier
            .fillMaxSize()
            .background(PrismaSemanticColors.SurfaceBase.themed()),
    ) {
        if (entry == null) EmptyDetail() else EntryDetail(entry = entry, scrollOffsets = scrollOffsets)
    }
}

@Composable
private fun EmptyDetail() {
    val total = CatalogueRegistry.entries.size
    val implemented = listOf(
        "foundation.typography", "foundation.colors", "foundation.spacing",
        "foundation.elevation",  "foundation.motion", "foundation.radius",
        "input.button",
    ).size

    Column(
        modifier = Modifier
            .fillMaxSize()
            .padding(PrismaSpacing.Sp7),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.Center,
    ) {
        // Subtle decorative ring — gives the empty state a centerpoint and brand presence.
        Box(
            modifier = Modifier
                .size(72.dp)
                .clip(CircleShape)
                .background(PrismaSemanticColors.AccentSubtle.themed()),
        )
        Text(
            text = "Prisma",
            style = PrismaTypography.DisplaySm,
            color = PrismaSemanticColors.TextPrimary.themed(),
            modifier = Modifier.padding(top = PrismaSpacing.Sp5),
        )
        Text(
            text = "Pick a component from the sidebar to see its showcase.",
            style = PrismaTypography.BodyLg,
            color = PrismaSemanticColors.TextSecondary.themed(),
            modifier = Modifier.padding(top = PrismaSpacing.Sp2),
        )

        // Stats strip — gives the empty state functional value.
        Row(
            modifier = Modifier.padding(top = PrismaSpacing.Sp7),
            horizontalArrangement = Arrangement.spacedBy(PrismaSpacing.Sp7),
        ) {
            Stat(label = "ENTRIES", value = total.toString())
            Stat(label = "IMPLEMENTED", value = implemented.toString())
            Stat(label = "PHASE", value = "1")
        }
    }
}

@Composable
private fun Stat(label: String, value: String) {
    Column(horizontalAlignment = Alignment.CenterHorizontally) {
        Text(
            text = value,
            style = PrismaTypography.HeadlineMd.copy(fontFamily = FontFamily.Monospace),
            color = PrismaSemanticColors.AccentDefault.themed(),
        )
        Text(
            text = label,
            style = PrismaTypography.LabelSm,
            color = PrismaSemanticColors.TextTertiary.themed(),
            modifier = Modifier.padding(top = PrismaSpacing.Sp1),
        )
    }
}

@Composable
private fun EntryDetail(
    entry: CatalogueEntry,
    scrollOffsets: androidx.compose.runtime.snapshots.SnapshotStateMap<String, Int>,
) {
    // Scroll state per-entry, owned by CatalogueShell so it survives the
    // AnimatedPane destruction on back-nav. Reads the stored pixel offset
    // for this entry on first composition, syncs back on every change.
    val scrollState = androidx.compose.runtime.remember(entry.key) {
        androidx.compose.foundation.ScrollState(scrollOffsets[entry.key] ?: 0)
    }
    androidx.compose.runtime.LaunchedEffect(entry.key, scrollState.value) {
        scrollOffsets[entry.key] = scrollState.value
    }
    Column(
        modifier = Modifier
            .fillMaxSize()
            .verticalScroll(scrollState)
            .padding(horizontal = PrismaSpacing.Sp8, vertical = PrismaSpacing.Sp7),
        verticalArrangement = Arrangement.spacedBy(PrismaSpacing.Sp4),
    ) {
        // Section breadcrumb pill.
        Box(
            modifier = Modifier
                .clip(RoundedCornerShape(PrismaRadius.Full))
                .background(PrismaSemanticColors.SurfaceSunken.themed())
                .padding(horizontal = PrismaSpacing.Sp3, vertical = PrismaSpacing.Sp1),
        ) {
            Text(
                text = entry.section.title,
                style = PrismaTypography.LabelSm,
                color = PrismaSemanticColors.TextSecondary.themed(),
            )
        }

        Text(
            text = entry.title,
            style = PrismaTypography.DisplaySm,
            color = PrismaSemanticColors.TextPrimary.themed(),
        )

        // Subtitle / status line.
        if (entry.content == null) {
            Text(
                text = "Spec ready · implementation pending in upcoming phase.",
                style = PrismaTypography.BodyMd,
                color = PrismaSemanticColors.TextTertiary.themed(),
            )
        }

        // Subtle separator before content.
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .height(1.dp)
                .padding(vertical = PrismaSpacing.Sp1)
                .background(PrismaSemanticColors.BorderSubtle.themed()),
        )

        val showcase = entry.content
        if (showcase != null) {
            Box(modifier = Modifier.padding(top = PrismaSpacing.Sp4)) {
                showcase()
            }
        } else {
            PlaceholderState(entry = entry)
        }
    }
}

@Composable
private fun PlaceholderState(entry: CatalogueEntry) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(PrismaRadius.Lg))
            .background(PrismaSemanticColors.SurfaceSunken.themed())
            .padding(PrismaSpacing.Sp7),
        verticalArrangement = Arrangement.spacedBy(PrismaSpacing.Sp3),
    ) {
        Text(
            text = "Implementation queued",
            style = PrismaTypography.TitleMd,
            color = PrismaSemanticColors.TextPrimary.themed(),
        )
        Text(
            text = "The spec for ${entry.title.lowercase()} is complete. Component code lands in the upcoming phase per docs/TODO.md.",
            style = PrismaTypography.BodyMd,
            color = PrismaSemanticColors.TextSecondary.themed(),
        )
        if (entry.tags.isNotEmpty()) {
            Row(horizontalArrangement = Arrangement.spacedBy(PrismaSpacing.Sp2)) {
                entry.tags.take(4).forEach { tag ->
                    TagChip(tag = tag)
                }
            }
        }
    }
}

@Composable
private fun TagChip(tag: String) {
    Box(
        modifier = Modifier
            .clip(RoundedCornerShape(PrismaRadius.Full))
            .background(PrismaSemanticColors.SurfaceRaised.themed())
            .padding(horizontal = PrismaSpacing.Sp3, vertical = PrismaSpacing.Sp1),
    ) {
        Text(
            text = tag,
            style = PrismaTypography.LabelSm,
            color = PrismaSemanticColors.TextSecondary.themed(),
        )
    }
}
